dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -1426,6 +1528,431 @@ async fn cmd_db_snapshot_info(
+                repair_id: notification.repair_id.to_string(),
+                repair_type: format!("{:?}", notification.repair_type),
+                upstairs_id: notification.upstairs_id.to_string(),
+                session_id: notification.session_id.to_string(),
+                notification_type: format!(
+                    "{:?}",
+                    notification.notification_type
+                ),
+            });
+        }
+
+        let table = tabled::Table::new(rows)
+            .with(tabled::settings::Style::empty())
+            .with(tabled::settings::Padding::new(0, 1, 0, 0))
+            .with(tabled::settings::Panel::header("Repair notifications"))
+            .to_string();
+
+        println!("{}", table);
+
+        println!();
+
+        // Use the most recent notification to get the most recent repair ID,
+        // and use that to search for progress.
+
+        let maybe_repair: Option<UpstairsRepairNotification> = datastore
+            .pool_connection_for_tests()
+            .await?
+            .transaction_async(|conn| async move {
+                use db::schema::upstairs_repair_notification::dsl;
+
+                dsl::upstairs_repair_notification
+                    .filter(dsl::region_id.eq(new_region_id))
+                    .filter(
+                        dsl::notification_type
+                            .eq(UpstairsRepairNotificationType::Started),
+                    )
+                    .order_by(dsl::time.desc())
+                    .limit(1)
+                    .first_async(&conn)
+                    .await
+                    .optional()
+            })
+            .await?;
+
+        if let Some(repair) = maybe_repair {
+            let maybe_repair_progress: Option<UpstairsRepairProgress> =
+                datastore
+                    .pool_connection_for_tests()
+                    .await?
+                    .transaction_async(|conn| async move {
+                        use db::schema::upstairs_repair_progress::dsl;
+
+                        dsl::upstairs_repair_progress
+                            .filter(dsl::repair_id.eq(repair.repair_id))
+                            .order_by(dsl::time.desc())
+                            .limit(1)
+                            .first_async(&conn)
+                            .await
+                            .optional()
+                    })
+                    .await?;
+
+            if let Some(repair_progress) = maybe_repair_progress {
+                let bar = ProgressBar::with_draw_target(
+                    Some(repair_progress.total_items as u64),
+                    ProgressDrawTarget::stdout(),
+                )
+                .with_style(ProgressStyle::with_template(
+                    "progress:\t{wide_bar:.green} [{pos:>7}/{len:>7}]",
+                )?)
+                .with_position(repair_progress.current_item as u64);
+
+                bar.abandon();
+
+                println!();
```

Comment:
The progress bar does not get to the end?
```
EVT22200005 # omdb db region-replacement info 3a4b04be-4bf9-43d6-a890-e5946d241084
note: database URL not specified.  Will search DNS.
note: (override with --db-url or OMDB_DB_URL)
note: using DNS server for subnet fd00:1122:3344::/48
note: (if this is not right, use --dns-server to specify an alternate DNS server)
note: using database URL postgresql://root@[fd00:1122:3344:101::4]:32221,[fd00:1122:3344:101::6]:32221,[fd00:1122:3344:101::5]:32221,[fd00:1122:3344:101::7]:32221,[fd00:1122:3344:101::3]:32221/omicron?sslmode=disable
note: database schema version matches expected (56.0.0)
      started: 2024-05-28 21:02:19.949975 UTC
        state: Complete
old region id: ff1e4d49-8959-42e1-9fcc-ca219a68ef8e
new region id: Some(897b3307-36a8-4d0f-8ea2-9ba1562e0898)

Repair notifications                                                                                                                                                        
TIME                           REPAIR_ID                            REPAIR_TYPE UPSTAIRS_ID                          SESSION_ID                           NOTIFICATION_TYPE 
2024-05-28 21:02:57.786939 UTC a661e5b9-0735-4ff6-a318-693349460b4b Live        23776670-5d21-449f-9552-efad0fe32b2a dd90808a-ff3b-41d0-a273-6a64e820458c Started           
2024-05-28 21:08:21.759356 UTC a661e5b9-0735-4ff6-a318-693349460b4b Live        23776670-5d21-449f-9552-efad0fe32b2a dd90808a-ff3b-41d0-a273-6a64e820458c Succeeded         

progress:        █████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████░ [    799/    800]

Repair steps                                                                                                                                
TIME                           STEP_TYPE DETAILS                                                                                            
2024-05-28 21:02:47.663609 UTC Propolis  instance Some(94122867-721b-40a2-aa85-518139367113) vmm Some(735975a0-26dd-4fe0-bb87-80e5770b16b4) 
```
Is there a way to short-circuit it if it is done?  Instead of printing the incomplete bar?

Reply:
I believe this is a bug in Crucible's notification logic, in that it doesn't send the progress update notification when it's done? I'd rather fix that bug than have omdb print the complete bar. Opened https://github.com/oxidecomputer/crucible/issues/1319